modules/performance/extension/siddhi-execution-performance/component/src/main/java/org/wso2/extension/siddhi/execution/throughput/CalculatePerformanceStreamProcessorExtension.java

```diff
@@ -58,16 +58,32 @@
 import java.util.concurrent.ExecutorService;
 
 /**
- * Input attributes to log is (iijTimeStamp (Long), value (Float)).
+ * Input attributes to log is (iijTimeStamp (Long), type (String) , Count(Integer) , Windowsize (Integer)).
  */
 @Extension(
         name = "throughput",
         namespace = "throughput",
         description = "Measuring performance of stream processor with simple passthrough",
         parameters = {
                 @Parameter(name = "iijtimestamp",
-                           description = "This value used to find the sending timestamp from client",
-                           type = {DataType.LONG}),
+                        description = "This value used to find the sending timestamp from client",
+                        type = {DataType.LONG}),
+
+                @Parameter(name = "type",
+                        description = "This value used to specify which metric value " +
+                                "you are going to exract(\"throughput\" " +
+                                "or \"latency\" or \"both\" )",
+                        type = {DataType.STRING}),
+
+                @Parameter(name = "windowsize",
+                        description = "This value used to determine for how " +
+                                "many events we are going to track the Metrics",
```

Comment:
For optional parameter put optional = true

Reply:
https://github.com/wso2/product-sp/pull/855/commits/3971112ad2ea08bef38437ea24bb12cad020692c